File: behavior_process/erlang/process.erl
start(Argumentos) ->              % Start the Server.
  spawn(server, init, [Argumentos]).

init(Argumentos) ->               % Initalize the internal process state.
  State = initialize_state(Argumentos),
  loop(State).

loop(State) ->                    % Receive and handle message.
  receive
    {handle, Msg} ->
      NewState = handle(Msg, State),
      loop(NewState);
    stop ->
      terminate(State)
  end.

terminate(State) ->               % Clean up prior to termination.
  clean_up(State).

File: behavior_process/elixir/process.ex
def start() do
  spawn(server, init, [arguments])
end

def init(arguments) do
  state = initialize_state(arguments),
  loop(State)
end

def loop(state) do
  receive do
    {handle, msg} ->
      newState = handle(msg, state),
      loop(newState)
    stop ->
      terminate(state)
  end
end

def terminate do
  clean_up(state)
end
